src/engine/comp/waypoint_graph.cpp

```diff
@@ -0,0 +1,124 @@
+#include "engine/vec.hpp"
+#include "waypoint_graph.hpp"
+#include <cmath>
+#include <iostream>
+#include <sparsepp/spp.h>
+#include <vector>
+
+void CompWaypointGraph::find_distances(
```

Comment:
Maybe add a comment here that this should be called once for all the waypoints to cache the distances - it wasn't immediately obvious, and I thought the function was instead calculating some distances & returning them